app/services/instagram/test_event_service.rb

```diff
@@ -0,0 +1,79 @@
+class Instagram::TestEventService
+  def initialize(messaging)
+    @messaging = messaging
+  end
+
+  def perform
+    Rails.logger.info("Processing Instagram test webhook event, #{@messaging}")
+
+    return false unless test_webhook_event?
+
+    create_test_text
+  end
+
+  private
+
+  def test_webhook_event?
+    @messaging[:sender][:id] == '12334' && @messaging[:recipient][:id] == '23245'
+  end
```

Comment:
Is this documented somewhere? Please provide links to the documentation where Meta suggests doing this check.

Reply:
This is the standard payload for test events. I assume they will send this same payload for verification. I will test this after we test the approval flow for self-hosted customers.
![CleanShot 2025-04-11 at 15 48 33@2x](https://github.com/user-attachments/assets/af278841-ad81-4fa1-a3f1-55c26833c5cd)